src/dsp_tools/commands/excel2json/json_header.py

```diff
@@ -0,0 +1,249 @@
+from dsp_tools.commands.excel2json.models.json_header import Keywords
+from dsp_tools.commands.excel2json.models.json_header import Prefixes
+from dsp_tools.commands.excel2json.models.json_header import Project
+from dsp_tools.commands.excel2json.models.json_header import User
+from dsp_tools.commands.excel2json.models.json_header import UserRole
+from dsp_tools.commands.excel2json.models.json_header import Users
+from dsp_tools.commands.excel2json.utils import check_contains_required_columns
+from dsp_tools.commands.excel2json.utils import check_required_values_get_position_in_excel
+from dsp_tools.commands.excel2json.utils import read_and_clean_all_sheets
+from dsp_tools.models.exceptions import InputError
+
+
+def get_json_header(excel_filepath: Path) -> JsonHeader:
+    """
+    Returns the header of the JSON file.
+    If an Excel file with the information is provided it is filled out.
+    Otherwise it will return a header with the fields left blank.
+
+    Args:
+        excel_filepath: path to the excel file
+
+    Returns:
+        JsonHeader object
+
+    Raises:
+        InputError: If the file does not conform to the specifics
+    """
+    if not excel_filepath.is_file():
+        print("No json_header.xlsx file found in the directory, empty header was added.")
+        return EmptyJsonHeader()
+    sheets_df_dict = read_and_clean_all_sheets(excel_filepath)
+    sheets_df_dict = {x.lower(): df for x, df in sheets_df_dict.items()}
+    if compliance_problem := _do_formal_compliance(sheets_df_dict):
+        raise InputError(compliance_problem.execute_error_protocol())
+    result = _process_file(sheets_df_dict)
+    if isinstance(result, ExcelFileProblem):
+        raise InputError(result.execute_error_protocol())
+    print("json_header.xlsx file used to construct the json header.")
+    return result
+
+
+def _do_formal_compliance(df_dict: dict[str, pd.DataFrame]) -> ExcelFileProblem | None:
+    expected_sheets = ["prefixes", "project", "description", "keywords"]
+
+    def _check_df(sheet: str) -> str | None:
+        if (df := df_dict.get(sheet)) is None:
+            return sheet
+        if len(df) == 0:
+            return sheet
+        return None
+
+    if empty := [x for x in expected_sheets if _check_df(x)]:
+        return ExcelFileProblem("json_header.xlsx", [EmptySheetsProblem(empty)])
+    return None
+
+
+def _process_file(df_dict: dict[str, pd.DataFrame]) -> ExcelFileProblem | FilledJsonHeader:
+    problems: list[Problem] = []
+    prefix_result = _do_prefixes(df_dict["prefixes"])
+    if isinstance(prefix_result, ExcelSheetProblem):
+        problems.append(prefix_result)
+    project_result = _do_project(df_dict)
+    if not isinstance(project_result, Project):
+        problems.extend(project_result)
+    if problems:
+        return ExcelFileProblem("json_header.xlsx", problems)
+    project = cast(Project, project_result)
+    prefixes = cast(Prefixes, prefix_result)
```

Comment:
there is a lot of type checking and casting going on. The reason for that is, that Python's type system is just not great... But I wonder, if many of those problems could be solved by separating validation from creation in these functions? Then you could first do all the validations; and when they are done, creation can be done less defensively because there should not be any errors anymore.

Reply:
I have thought about that too, the reason why I combined it is because if I do not i need to write the code that retrieves the information twice... It felt more cumbersome to me. But I don't think I would like either one version 100%

Reply:
in the end it's up to you. But to me the retrieving code seems so minimal that I don't think the duplication would hurt.